plugins/home/backstage-plugin-home-markdown/src/MarkdownCard/Content.tsx

```diff
@@ -14,34 +14,80 @@
  * limitations under the License.
  */
 
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Alert } from '@material-ui/lab';
 import { Progress, MarkdownContent } from '@backstage/core-components';
-import { useGithubFile } from './useGithubFile';
 import {
   useApi,
   githubAuthApiRef,
   SessionState,
+  useApiHolder,
+  ApiHolder,
 } from '@backstage/core-plugin-api';
 import { MarkdownContentProps } from './types';
 import { Button, Grid, Typography, Tooltip } from '@material-ui/core';
+import useAsync from 'react-use/lib/useAsync';
+import { githubApiRef, GithubClient, GithubApi } from '../apis';
+
+const getGithubClient = (apiHolder: ApiHolder) => {
+  let githubClient: GithubApi | undefined = apiHolder.get(githubApiRef);
+  if (!githubClient) {
+    const auth = apiHolder.get(githubAuthApiRef);
+    if (auth) {
+      githubClient = new GithubClient({ githubAuthApi: auth });
+    }
+  }
+  if (!githubClient) {
+    throw new Error(`Failed to get the github client`);
+  }
+  return githubClient;
+};
 
 const GithubFileContent = (props: MarkdownContentProps) => {
-  const { value, loading, error } = useGithubFile({ ...props });
+  const { dontStripHtmlCommentsBeforeRendering } = props;
+  const apiHolder = useApiHolder();
+
+  const { value, loading, error } = useAsync(async () => {
+    const githubClient = getGithubClient(apiHolder);
+    return githubClient.getContent({ ...props });
+  });
+
+  const transformImageUri = useCallback(
+    (href: string) => {
+      return value?.media[href] || href;
+    },
+    [value?.media],
+  );
+
+  const transformLinkUri = useCallback(
+    (href: string) => {
+      return value?.links[href] || href;
+    },
+    [value?.links],
+  );
 
   if (loading) {
     return <Progress />;
   } else if (error) {
     return <Alert severity="error">{error.message}</Alert>;
   }
 
-  let content = Buffer.from(value.content, 'base64').toString('utf8');
+  if (!value) {
+    return <Progress />;
+  }
 
-  if (props.purgeHtmlComments) {
+  let content = value.content;
+  if (!dontStripHtmlCommentsBeforeRendering) {
```

Comment:
maybe something like `preserveHtmlComments` or just defaulting the original to false might be better here to avoid the double negation. Also easier to default things to false instead of true